torchgeo/datasets/soda.py

```diff
@@ -0,0 +1,362 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+"""SODA datasets."""
+
+import json
+import os
+from collections.abc import Callable
+from typing import ClassVar
+
+import matplotlib.patches as patches
+import matplotlib.pyplot as plt
+import numpy as np
+import pandas as pd
+import torch
+from matplotlib.figure import Figure
+from PIL import Image
+from shapely.geometry import MultiPoint, Polygon
+from torch import Tensor
+
+from .errors import DatasetNotFoundError
+from .geo import NonGeoDataset
+from .utils import Path, check_integrity, download_and_extract_archive, download_url
+
+
+class SODAA(NonGeoDataset):
+    """SODA-A dataset.
+
+    The `SODA-A <https://shaunyuan22.github.io/SODA/>`_ dataset is a high resolution aerial imagery dataset for small object detection.
+
+    Dataset features:
+
+    * 2513 images
+    * 872,069 annotations with oriented bounding boxes
+    * 9 classes
+
+    Dataset format:
+
+    * Images are three channel .jpg files.
+    * Annotations are in json files
+
+    Classes:
+
+    0. Airplane
+    1. Helicopter
+    2. Small vehicle
+    3. Large vehicle
+    4. Ship
+    5. Container
+    6. Storage tank
+    7. Swimming-pool
+    8. Windmill
+    9. Other
+
+    If you use this dataset in your research, please cite the following paper:
+
+    * https://ieeexplore.ieee.org/document/10168277
+
+    .. versionadded:: 0.7
+    """
+
+    url = 'https://hf.co/datasets/torchgeo/soda-a/resolve/b082b9555ea9960d614b54a8cecde4cc63ec5481/{}'
+
+    files: ClassVar[dict[str, dict[str, str]]] = {
+        'images': {
+            'filename': 'Images.zip',
+            'md5sum': '8ee4ad7a306b0a0a900fa78a4f6aae68',
+        },
+        'labels': {
+            'filename': 'Annotations.zip',
+            'md5sum': '45b0d21209fc332d89b0144b308e57fa',
+        },
+    }
+
+    classes = (
+        'airplane',
+        'helicopter',
+        'small-vehicle',
+        'large-vehicle',
+        'ship',
+        'container',
+        'storage-tank',
+        'swimming-pool',
+        'windmill',
+        'other',
+    )
+
+    valid_splits = ('train', 'val', 'test')
+
+    valid_orientations = ('oriented', 'horizontal')
+
+    def __init__(
+        self,
+        root: Path = 'data',
+        split: str = 'train',
+        bbox_orientation: str = 'oriented',
```

Comment:
```suggestion
        split: Literal['train', 'val', 'test'] = 'train',
        bbox_orientation: Literal['oriented', 'horizontal'] = 'oriented',
```
I'm trying to push us to start using `typing.Literal` more so we can more clearly document and test valid values.